File: frontend/src/modules/proyectos/application/CrearProyectoUseCase.ts
import { ProyectoRepositoryPort } from '../domain/ProyectoRepositoryPort';
import { CreateProyectoDto } from '../domain/CreateProyectoDto';
import { Proyecto } from '../domain/Proyecto';

export class CrearProyectoUseCase {
  constructor(private repository: ProyectoRepositoryPort) {}

  async ejecutar(dto: CreateProyectoDto): Promise<Proyecto> {
    if (!dto.name.trim()) throw new Error('El nombre del proyecto es obligatorio.');
    return this.repository.crear(dto);
  }
}

File: backend/modulos/proyectos/aplicacion/CrearProyectoUseCase.py
from modulos.proyectos.dominio.ProyectoRepositoryPort import ProyectoRepositoryPort
from modulos.proyectos.dominio.Entidades import Proyecto

class CrearProyectoUseCase:
    def __init__(self, repositorio: ProyectoRepositoryPort):
        self.repositorio = repositorio
    def ejecutar(self, name: str, created_by_id: int, **kwargs) -> Proyecto:
        return self.repositorio.crear(Proyecto.crear(name=name, created_by_id=created_by_id, **kwargs))
